### DISCUSSION TITLE
Start jobs from job or sensor with dynamical configuration

### DISCUSSION THREAD

**Discussion by mmarionncern** [dagster-io/dagster#9589] [Q&A] [answered]
Dear dagster devs and experts,

I am looking for a way to launch several jobs from either a sensor or more suitably a job for which I can change easily the configuration as function of the jobs I want to launch in dagit, e.g this kind of configuration :

job_list: 
 - job_name: job_1
   job_configs: [ config_1 ]
 - job_name: job_2
   job_configs: [ config_2_a, config_2_b]

that would launch three jobs on two different kinds with different configuration.

What would be the best way to do so?

Thank you by advance!

**Comment by yuhan** [comment, marked as answer]
You can programmatically execute jobs using the [`execute_job`](https://docs.dagster.io/_apidocs/execution#dagster.execute_job) Python API:
```python
@job
def job_1():
    ...

@job
def job_2():
    ...


instance = DagsterInstance.get()
execute_job(reconstructable(job_1), instance=instance, run_config=config_1)
execute_job(reconstructable(job_2), instance=instance, run_config=config_2_a)
execute_job(reconstructable(job_2), instance=instance, run_config=config_2_b)
```

So, you could build a "master" job that calls the `execute_job` under the hood to kick off other jobs. Something like:
```python
from dagster import Field, repository, op, job, execute_job, reconstructable, Array, Shape


@job
def job_1():
    ...


@job
def job_2():
    ...


JOB_NAME_TO_DEF = {
    "job_1": job_1,
    "job_2": job_2,
}


@op(
    config_schema={
        "ls_of_job_name_and_config": Field(Array(Shape({"job_name": str, "job_config": dict})))
    }
)
def kick_off_jobs(context):
    ls_of_job_name_and_config = context.op_config["ls_of_job_name_and_config"]
    for entry in ls_of_job_name_and_config:
        job_name = entry["job_name"]
        job_config = entry["job_config"]
        job_def = JOB_NAME_TO_DEF[job_name]
        print(f"executing {job_name} with config: ", job_config)

        execute_job(
            reconstructable(job_def), instance=context.instance, run_config=job_config,
        )


@job
def master_job():
    kick_off_jobs()


@repository
def my_repo():
    return [master_job, job_1, job_2]

```
And then, you can provide configs via your Dagit launchpad to kick off runs:

<img width="1648" alt="Screen Shot 2022-09-07 at 2 36 56 PM" src="https://user-images.githubusercontent.com/4531914/188986804-12753697-2918-4c65-be10-e7938396bb00.png">

You can find more available config schema here: https://docs.dagster.io/_apidocs/config#dagster.ConfigSchema

**Comment by mmarionncern** [comment]
Thanks a lot!
 this is exactly what I was looking for.

**Comment by mmarionncern** [comment]
To go further, is that easy to create new instances in order to efficienctly run jobs asynchronously on the same logic?

Thanks!

**Reply by yuhan** [reply]
Yes, I believe you can leverage [Dynamic Graphs](https://docs.dagster.io/concepts/ops-jobs-graphs/dynamic-graphs) to execute one job per dynamic op. Ops will be run in multi-processing async by default.
